### PR TITLE
Add modulo operator

### PR DESCRIPTION
**Pull request by mattnibs**
Add modulo operator for binary expressions. Also get rid of mod function
since this the modulo operator should be used instead.

### CODE REVIEW

File: expr/eval.go [hunk 1]
```diff
@@ -548,6 +554,32 @@ func (d *Divide) Eval(rec *zed.Value) (zed.Value, error) {
 	return zed.Value{}, ErrIncompatibleTypes
 }
 
+func (m *Modulo) Eval(zv *zed.Value) (zed.Value, error) {
+	id, err := m.eval(zv)
+	if err != nil {
+		fmt.Println("returning.err")
```
Comment (nwt) [inline expr/eval.go:560]:
```suggestion
```

File: expr/ztests/operaor-mod.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  {a:5,b:0}
+
+output: |
+  {res:1}
+  {res:1}
+  {res:1}
+  {res:1(uint64)}
+  {res:"operator % not defined on type float64"(error)}
+  {res:"divide by zero"(error)}
+
```
Comment (nwt) [inline expr/ztests/operaor-mod.yaml:18]:
```suggestion
```

File: expr/eval.go [hunk 3]
```diff
@@ -548,6 +554,32 @@ func (d *Divide) Eval(rec *zed.Value) (zed.Value, error) {
...
+		y, _ := zed.DecodeInt(m.vals.B)
+		if y == 0 {
+			return zed.NewErrorf("divide by zero"), nil
+		}
+		return zed.Value{typ, m.vals.Int(x % y)}, nil
+	}
+	x, _ := zed.DecodeUint(m.vals.A)
+	y, _ := zed.DecodeUint(m.vals.B)
+	if y == 0 {
+		return zed.NewErrorf("divide by zero"), nil
```
Comment (nwt) [inline expr/eval.go:578]:
```suggestion
			return zed.NewErrorf("modulo by zero"), nil
		}
		return zed.Value{typ, m.vals.Int(x % y)}, nil
	}
	x, _ := zed.DecodeUint(m.vals.A)
	y, _ := zed.DecodeUint(m.vals.B)
	if y == 0 {
		return zed.NewErrorf("modulo by zero"), nil
```

**Review by nwt — APPROVED** [review]

**Comment by jameskerr** [comment]
**Verified.  This works as expected.** 

This is the output of the tests I ran.

**n % 2** 
```
$ 1 echo '{n: 1} {n:2} {n:3}' | zq 'put r := (n % 2)' -
{n:1,r:1}
{n:2,r:0}
{n:3,r:1}
```

**n % -2** 
```
$ echo '{n: 1} {n:2} {n:3}' | zq 'put r := (n % -2)' -
{n:1,r:1}
{n:2,r:0}
{n:3,r:1}
```

**n % -10* 
```
$ echo '{n: 1} {n:2} {n:3}' | zq 'put r := (n % -10)' -
{n:1,r:1}
{n:2,r:2}
{n:3,r:3}
```

**n % -0.5* 
```
$ echo '{n: 1} {n:2} {n:3}' | zq 'put r := (n % -0.5)' -
{n:1,r:"operator % not defined on type float64"(error)}
{n:2,r:"operator % not defined on type float64"(error)}
{n:3,r:"operator % not defined on type float64"(error)}
```

**Non Integer**

```
$ echo '{n: 1} {n:2} {n:3} {n: "NaN"}' | zq 'put r := (n % 2)' -
incompatible types
{n:1,r:1}
{n:2,r:0}
{n:3,r:1}
{n:"NaN"}
```

**By zero**
```
$ echo '{n: 1} {n:2} {n:3} {n: "NaN"}' | zq 'put r := (n % 0)' -
incompatible types
{n:1,r:"modulo by zero"(error)}
{n:2,r:"modulo by zero"(error)}
{n:3,r:"modulo by zero"(error)}
{n:"NaN"}
```

**Ensure the function is gone**
```
$ echo '{n: 1} {n:2} {n:3} {n: "NaN"}' | zq 'put r := mod(this.n, 0)' -

rhs of assigment expression: mod(): no such function
```

**Pull request merged**